### ISSUE TITLE
Pattern Matching with type()

### ISSUE THREAD

**Issue by KalleDK** [microsoft/pyright#5573]
## Environment data
-   Language Server version:  2023.7.31 (pyright b917b7a5)
-   OS and version: WSL2 Debian
-   Python version (& distribution if applicable, e.g. Anaconda): 3.11

## Code Snippet

```python
y = str

match y:
    case type(): # <-- This line complains - Pattern will never be matched for subject type "y"
        print(8)
    case _:
        pass
```

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.319, which I just published. It will also be included in a future release of pylance.
